docs/content/docs/dev/table/sql/queries/vector-search.md

```diff
@@ -0,0 +1,115 @@
+---
+title: "Vector Search"
+weight: 7
+type: docs
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Vector Search
+
+{{< label Batch >}} {{< label Streaming >}}
+
+Flink SQL provides the `VECTOR_SEARCH` table-valued function (TVF) to perform vector search in SQL queries. This function allows you to search similar rows according to the high-dimension vectors.
+
+## VECTOR_SEARCH Function
+
+The `VECTOR_SEARCH` uses a processing-time attribute to correlate rows to the latest version of data in an external table. It's very similar to lookup join in Flink SQL, however, the difference is 
+`VECTOR_SEARCH` uses the input data vector to compare the similarity among data in the external table and return the top-k most similar rows.
+
+### Syntax
+
+```sql
+SELECT * 
+FROM input_table, LATERAL TABLE(VECTOR_SEARCH(
+   TABLE vecotr_table, 
+   input_table.vector_column, 
+   DESCRIPTOR(index_column),
+   top_k,
+   [CONFIG => MAP['key', 'value']]
+   ))
+```
+
+### Parameters
+
+- `input_table`: The input table containing the data to be processed
+- `vector_table`: The name of external table that allows to search via vector
+- `vector_column`: The name of the columns in the input table, its type should be FLOAT ARRAY or DOUBLE ARRAY
+- `index_column`: A descriptor specifying which columns from the vector table should be used to compare the similarity with the input data
```

Comment:
index_column is singular. should `which columns` be which `column`.

Or does it allow multiple column inputs ?